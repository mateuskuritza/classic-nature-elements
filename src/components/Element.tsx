import styled from "styled-components";

export default function Element({ name }: { name: string }) {
	return <Image src={`./assets/${name}.jpg`} alt={`${name} element`} />;
}

const Image = styled.img`
	height: 100px;
	width: 100px;
	object-fit: cover;
	border-radius: 100%;
	border: 1px solid black;
`;
