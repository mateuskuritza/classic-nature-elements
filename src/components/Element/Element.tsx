import styled from "styled-components";
import IElement from "../../interfaces/IElement";
import Button from "../Button/Button";
import Learned from "../Learned/Learned";
import NotLearned from "../NotLearned/NotLearned";

export default function Element({ elementInfos: { name, learned }, changeElement }: { elementInfos: IElement; changeElement: CallableFunction }) {
	return (
		<Container>
			<Image src={`./assets/${name}.jpg`} alt={`${name} element`} />
			<Button onClick={() => changeElement(name, !learned)}>{learned ? "Esquecer" : "Dominar"}</Button>
			<p>{name.toUpperCase()}</p>
			{learned ? <Learned /> : <NotLearned />}
		</Container>
	);
}

const Container = styled.div`
	width: 150px;
	height: 250px;
	padding: 10px 6px;

	border: 1px solid #969696;

	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: space-between;
`;

const Image = styled.img`
	height: 100px;
	width: 100px;
	object-fit: cover;
	border-radius: 100%;
	border: 1px solid black;
`;
