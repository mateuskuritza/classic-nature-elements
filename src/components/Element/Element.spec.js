import React from "react";
import { render, screen } from "@testing-library/react";
import Element from "./Element";

it("Should render Element", () => {
    const element = {
        name: 'fire',
        learned: false
    }
    const { container } = render(<Element elementInfos={element} />);
    expect(container).toMatchSnapshot();
});

it("Should render name upperCase", () => {
    const element = {
        name: 'fire',
        learned: false
    }
    const { container } = render(<Element elementInfos={element} />);
    const nameText = screen.getByText(element.name.toUpperCase())
    expect(nameText).toBeDefined();
    expect(container).toMatchSnapshot();
});

it("Should render Dominar text and NotLearned component", () => {
    const element = {
        name: 'fire',
        learned: false
    }
    const { container } = render(<Element elementInfos={element} />);

    const buttonTextNotLearned = screen.getByText('Dominar');
    expect(buttonTextNotLearned).toBeDefined();

    const notLearnedComponent = screen.getByText('NÃ£o dominado');
    expect(notLearnedComponent).toBeDefined();

    expect(container).toMatchSnapshot();
});

it("Should render Esquecer text and Learned component", () => {
    const element = {
        name: 'fire',
        learned: true
    }
    const { container } = render(<Element elementInfos={element} />);

    const buttonTextLearned = screen.getByText('Esquecer');
    expect(buttonTextLearned).toBeDefined();

    const learnedComponent = screen.getByText('Dominado!');
    expect(learnedComponent).toBeDefined();

    expect(container).toMatchSnapshot();
});

it("Should correct img (src) and alt text", () => {
    const element = {
        name: 'fire',
        learned: true
    }
    const { container } = render(<Element elementInfos={element} />);

    const img = screen.getByRole('img');
    expect(img.getAttribute('src')).toBe(`./assets/${element.name}.jpg`);
    expect(img.getAttribute('alt')).toBe(`${element.name} element`);

    expect(container).toMatchSnapshot();
});