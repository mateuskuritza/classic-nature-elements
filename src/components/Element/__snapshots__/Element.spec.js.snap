// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correct img (src) and alt text 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Esquecer
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-gsDKAQ fjiGyX"
    >
      <p>
        Dominado!
      </p>
    </div>
  </div>
</div>
`;

exports[`Should render Dominar text and NotLearned component 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Dominar
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-dkPtRN iRVeL"
    >
      <p>
        Não dominado
      </p>
    </div>
  </div>
</div>
`;

exports[`Should render Element 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Dominar
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-dkPtRN iRVeL"
    >
      <p>
        Não dominado
      </p>
    </div>
  </div>
</div>
`;

exports[`Should render Esquecer text and Learned component 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Esquecer
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-gsDKAQ fjiGyX"
    >
      <p>
        Dominado!
      </p>
    </div>
  </div>
</div>
`;

exports[`Should render name correct img src and alt text 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Esquecer
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-gsDKAQ fjiGyX"
    >
      <p>
        Dominado!
      </p>
    </div>
  </div>
</div>
`;

exports[`Should render name upperCase 1`] = `
<div>
  <div
    class="sc-hKwDye IcbJx"
  >
    <img
      alt="fire element"
      class="sc-eCImPb grWgTS"
      src="./assets/fire.jpg"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Dominar
    </button>
    <p>
      FIRE
    </p>
    <div
      class="sc-dkPtRN iRVeL"
    >
      <p>
        Não dominado
      </p>
    </div>
  </div>
</div>
`;
